js/chips.js

```diff
@@ -1,16 +1,22 @@
-// some mock functions
+var selectedChip; // creates a global variable to track the selectedChip element without requiring looping through the DOM for classes
 
 function selectChip(chip) {
-	// make sure any other selected chip on the board is unselected.
-	// Only one chip should be selected at any time.
-	clearSelection();
+
+	// Only one chip should be selected at any time, so if the selectedChip variable is truthy, we should clear that.
+	if(selectedChip) {
+		clearSelection();
+	}
```

Comment:
now that we have `selectedChip` as a global var, we don't need to loop through the DOM to find any chip that has the `.selected` class. We only need to evaluate the `selectedChip` var. This is more performant! Thus we only need to `clearSelection` if the `selectedChip` exists.

Reply:
makes so much more sense this way